runtime/nativeMain/src/kotlinx/benchmark/native/NativeExecutor.kt

```diff
@@ -96,24 +197,30 @@ class NativeExecutor(name: String, args: Array<out String>) : SuiteExecutor(name
         name: String,
         config: BenchmarkConfiguration,
         instance: T,
-        benchmark: BenchmarkDescriptor<T>
+        benchmark: BenchmarkDescriptor<T>,
+        currentIteration: Int? = null
     ): Int {
         var iterations = 0
-        repeat(config.warmups) { iteration ->
+        val warmupIterations = if (config.iterationMode == IterationMode.External) 1 else config.warmups
```

Comment:
Is one warmup iteration enough to get a reasonable measurement cycles number?
And is it enough for caches warmup?

Reply:
Warm up is already done, for cycles number it should be enough.

Reply:
May be logic of getting cycles number is worth to include inside iteration.

Reply:
What about passing the number of cycles from actual warmup iterations? (e.g. through a temp file)
What bothers me is that a measurement iteration takes two times longer that it is supposed to take. It may be significant for long-running benchmarks, nightly runs

Reply:
If warmup iteration time difers from measurement iterations time, the number could be adjusted by the measurement_time/warmup_time ratio. 
Alternatively, it is possible to run an initial iteration with measurement iterations time for calculating cycles without reporting the result of that iteration, and use the calculated cycles to run measurement iterations.